### PR TITLE
ServerConfig tests

### PR DESCRIPTION
**Pull request by KyCodeHuynh**
This adds unit tests for [ServerConfig](https://github.com/UCLA-CS130/Mr.-Robot-et-al./blob/master/server_config.h).

### CODE REVIEW

File: server_config_test.cc [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    return std::unique_ptr<ServerConfig>(new ServerConfig(out_config_));
+  }
+
+  NginxConfigParser parser_;
+  NginxConfig out_config_;
+};
+
+TEST_F(ServerConfigTest, NoPortConfig) {
+    std::unique_ptr<ServerConfig> server_config = ParseString("server { ACM Hack; }");
+    EXPECT_EQ("", server_config->propertyLookUp("port"));
```
Comment (apfong) [inline server_config_test.cc]: What's are the convention for testing with GTest in regard to outputting a message when the assert fails? I know before we had something like `EXPECT_EQ() << "msg";` for each test in the first assignment

File: server_config_test.cc [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    return std::unique_ptr<ServerConfig>(new ServerConfig(out_config_));
+  }
+
+  NginxConfigParser parser_;
+  NginxConfig out_config_;
+};
+
+TEST_F(ServerConfigTest, NoPortConfig) {
+    std::unique_ptr<ServerConfig> server_config = ParseString("server { ACM Hack; }");
```
Comment (apfong) [inline server_config_test.cc]: Hack is back.

File: server_config_test.cc [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+TEST_F(ServerConfigTest, HighlyNestedPropertiesSet) {
+    std::unique_ptr<ServerConfig> server_config = ParseString("outer { server { echo true; listen 1234; } }");
+    EXPECT_EQ("1234", server_config->propertyLookUp("port"));
+}
+
+TEST_F(ServerConfigTest, PropertiesSetWithNewlines) {
+    std::unique_ptr<ServerConfig> server_config = ParseString("server { echo \ntrue; listen \n1234; }");
+    EXPECT_EQ("1234", server_config->propertyLookUp("port"));
+}
```
Comment (apfong) [inline server_config_test.cc]: Should we have a test for if our config file has just `listen` or `server { listen }`?

Comment (KyCodeHuynh) [inline server_config_test.cc]: Good thinking! Added as part of https://github.com/UCLA-CS130/Mr.-Robot-et-al./commit/714a7316556e8ed83136c42e2eae6843c9579f7c

File: server_config_test.cc (hunk 1 above)
Comment (KyCodeHuynh) [inline server_config_test.cc]: Nice catch, I forgot to include fail-case messages. I've added them in https://github.com/UCLA-CS130/Mr.-Robot-et-al./commit/a793e20edc43ce7bd2b33372a4fee553851709c6

File: server_config.cc [hunk 4]
```diff
@@ -21,9 +21,11 @@ void ServerConfig::fillOutMap(NginxConfig config) {
       fillOutMap(*(config.statements_[i]->child_block_));
     }
 
-    property = config.statements_[i]->tokens_[0];
```
Comment (apfong) [inline server_config.cc]: Good catch

**Pull request merged**